x/confidentialtransfers/keeper/msg_server.go

```diff
@@ -242,6 +254,14 @@ func (m msgServer) Withdraw(goCtx context.Context, req *types.MsgWithdraw) (*typ
 		return nil, sdkerrors.Wrapf(sdkerrors.ErrInsufficientFunds, "insufficient funds to withdraw %s %s", req.Amount, req.Denom)
 	}
 
+	gasSoFar := ctx.GasMeter().GasConsumed()
+	multiplier := m.Keeper.GetRangeProofGasMultiplier(ctx)
+
+	// Consume additional gas according to the multiplier as range proofs are computationally expensive.
+	if multiplier > 1 {
+		ctx.GasMeter().ConsumeGas(gasSoFar*uint64(multiplier-1), "range proof verification")
```

Comment:
curious why are we decrementing multiplier?

Reply:
Multiplier tells us how much to multiply the gas cost by - gasSoFar represents the amount of gas the tx would normally spend without any manipulation.
So let's say multiplier is set to 10, then gas cost will be 

gasSoFar + (9 x gasSoFar) = 10 x gasSoFar

Reply:
Added comment here to explain the calculation